modules/gcloud/gcloud.go

```diff
@@ -69,6 +71,22 @@ func WithProjectID(projectID string) Option {
 	}
 }
 
+// WithDataYamlFile seeds the Bigquery project for the GCloud container.
+func WithDataYamlFile(dataYamlFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
```

Comment:
Sorry @mdelapenya I recreated the PR to try to get the updates since I had last forked the project. I think I've made the error updates requested and updated the BQ docs to include information on this function.

Reply:
Hey @mtellis2 sorry for the late response. I'm taking this PR and resolving conflicts after the merge with main. I also addressed @stevenh's comments about using require and the new cleanup APIs for tests.

I have one concern regarding the functional option: if a user passes the option multiple times, it will be appending multiple files to be copied before the container is started, but overriding the file in the container (see https://github.com/testcontainers/testcontainers-go/pull/2523/files#diff-658ed21fe179e3c4810748c731c20b75757c269dd8ce30f0bf98ce31f2addd24R79), which means that the last added YAML file wins. Same for the CMD, as we are appending a new entry to the CMD args. I verified it's harmless regarding goccy's emulator to receive multiple values for the same flag, but then I verified that, if another data yaml file is added, the latest added yaml file wins.

To resolve it, I did a quick test and changed the option to support multiple files to be added, and the emulator just uses the last `--data-from-yaml` flag value. So I'm changing the implementation to support just one YAML file, adding a note in the docs for the functional option regarding this.

Hope you can take a look at the reviewed PR with my commits on top of yours.